crawl-ref/source/menu.cc

```diff
@@ -249,6 +249,11 @@ void Menu::set_flags(int new_flags, bool use_options)
     if (use_options && Options.easy_exit_menu)
         flags |= MF_EASY_EXIT;
 
+#ifdef DEBUG
+    int sel_flag = flags & (MF_NOSELECT | MF_SINGLESELECT | MF_MULTISELECT);
+    ASSERT(sel_flag == MF_NOSELECT || sel_flag == MF_SINGLESELECT || sel_flag == MF_MULTISELECT);
+#endif
+
```

Comment:
Any particular reason this is debug-build specific?  We usually (though not always) have `ASSERT`s be production.

Reply:
I think my reasoning was that this would prevent a spurious unused variable warning if `ASSERT`s were disabled